envi/archs/h8/disasm.py

```diff
@@ -0,0 +1,680 @@
+
+        mcanv.addNameText(mnem, typename="mnemonic")
+        mcanv.addText(" ")
+
+        # Allow each of our operands to render
+        imax = len(self.opers)
+        lasti = imax - 1
+        for i in xrange(imax):
+            oper = self.opers[i]
+            oper.render(mcanv, self, i)
+            if i != lasti:
+                mcanv.addText(",")
+
+    def __repr__(self):
+        mnem = self.mnem
+        if self.iflags & IF_B:
+            mnem += '.b'
+        elif self.iflags & IF_W:
+            mnem += '.w'
+        elif self.iflags & IF_L:
+            mnem += '.l'
+        
+        x = []
+        for o in self.opers:
+            x.append(o.repr(self))
+
+        return mnem + " " + ", ".join(x)
+
+
+
+def unittest_parsers(buf = 'ABCDEFGHIJKLMNOP', off=3, va=0x2544):
+    val, = struct.unpack('>H', buf[off:off+2])
+
+    for tsize in (1,2,4):
+        p_i3_Rd(va, val, buf, off, tsize)
+        p_i3_aERd(va, val, buf, off, tsize) 
+        p_i3_aAA8(va, val, buf, off, tsize) 
+        p_i8_CCR(va, val, buf, off, tsize) 
+        p_i8_Rd(va, val, buf, off, tsize) 
+        p_i16_Rd(va, val, buf, off, tsize) 
+        p_i32_ERd(va, val, buf, off, tsize) 
+        p_Rd(va, val, buf, off, tsize) 
+        p_Rs_Rd(va, val, buf, off, tsize)  
+        p_Rs_Rd_4b(va, val, buf, off, tsize)  
+        p_Rs_ERd(va, val, buf, off, tsize)  
+        p_Rs_ERd_4b(va, val, buf, off, tsize)  
+        p_ERd(va, val, buf, off, tsize)  
+        p_ERs_ERd(va, val, buf, off, tsize)  
+        p_Rn_Rd(va, val, buf, off, tsize)  
+        p_Rn_aERd(va, val, buf, off, tsize)  
+        p_Rn_aAA8(va, val, buf, off, tsize)  
+        p_aERn(va, val, buf, off, tsize)  
+        p_aAA24(va, val, buf, off, tsize)  
+        p_aaAA8(va, val, buf, off, tsize)  
+        p_1_Rd(va, val, buf, off, tsize)  
+        p_2_Rd(va, val, buf, off, tsize)  
+        p_4_Rd(va, val, buf, off, tsize)  
+        p_1_ERd(va, val, buf, off, tsize)  
+        p_2_ERd(va, val, buf, off, tsize)  
+        p_4_ERd(va, val, buf, off, tsize)  
+        p_disp8(va, val, buf, off, tsize)  
+        p_dis16(va, val, buf, off, tsize)  
+        p_nooperands(va, val, buf, off, tsize) 
+
+class H8Operand(envi.Operand):
+    tsize = 2
+    def involvesPC(self):
+        return False
+
+class H8RegDirOper(envi.RegisterOper, H8Operand):
+    ''' 
+    Register direct [Rn]
+    '''
+
+    def __init__(self, reg, tsize=4, va=0, oflags=0):
+        self.va = va
+        self.reg = convertMeta(reg, tsize)
+        self.tsize = tsize
+        self.oflags = oflags
+
+    def __eq__(self, oper):
+        if not isinstance(oper, self.__class__):
+            return False
+        if self.reg != oper.reg:
+            return False
+        if self.tsize != oper.tsize:
+            return False
+        if self.oflags != oper.oflags:
+            return False
+        return True
+    
+    def involvesPC(self):
+        return self.reg == REG_PC
+
+    def isDeref(self):
```

Comment:
For any of these methods ( and the ones in classes below ) where over-ridden method here does nothing other than what the base class does, the over-rides should be removed. ( less cruft, way easier to maintain, and basically what class inheritance is for :) )